x-pack/plugins/cases/server/client/cases/utils.ts

```diff
@@ -453,3 +457,30 @@ export const getUserProfiles = async (
     return acc;
   }, new Map());
 };
+
+export const validateCustomFieldKeysAgainstConfiguration = ({
+  requestCustomFields,
+  configurationCustomFields = [],
+}: {
+  requestCustomFields: CaseRequestCustomFields;
+  configurationCustomFields?: CustomFieldsConfiguration;
+}): string[] => {
+  const invalidCustomFieldKeys: string[] = [];
+
+  requestCustomFields.forEach((customField) => {
+    let validKey = false;
+    configurationCustomFields.every(({ key: keyInConfiguration }) => {
+      if (keyInConfiguration === customField.key) {
+        validKey = true;
+      }
+
+      return !validKey;
+    });
+
+    if (!validKey) {
+      invalidCustomFieldKeys.push(customField.key);
+    }
+  });
+
+  return invalidCustomFieldKeys;
```

Comment:
nit: To simplify the code you can use the `difference` helper method of `lodash`.

Reply:
Alternative you can do:

```
const customFieldsKeys = requestCustomFields.map((customField) => customField.key)
const configurationKeys = configurationCustomFields.map((conf) => conf.key)
const diff = customFieldsKeys.filter(customField => !configurationKeys.includes(x));

return diff;
```

Reply:
Much better now, I never remember to use `lodash` 😬 

```
return differenceWith(
  requestCustomFields,
  configurationCustomFields,
  (arrVal, othVal) => arrVal.key === othVal.key
).map((e) => e.key);
```